server/channels/api4/emoji.go

```diff
@@ -232,6 +239,36 @@ func getEmojiByName(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func getEmojisByNames(c *Context, w http.ResponseWriter, r *http.Request) {
+	names := model.ArrayFromJSON(r.Body)
+	if len(names) == 0 {
+		c.SetInvalidParam("names")
+		return
+	}
+
+	if !*c.App.Config().ServiceSettings.EnableCustomEmoji {
+		c.Err = model.NewAppError("getEmojisByNames", "api.emoji.disabled.app_error", nil, "", http.StatusNotImplemented)
+		return
+	}
```

Comment:
Apologies for the drive-by comment, but `App.GetMultipleEmojiByName` already checks for this.

Reply:
@agnivade Huh, I thought these checks tended to be on the API layer and not the app layer. Any opposition to leaving the duplicate check in?

Reply:
Can we remove the check in the `app` layer?

Reply:
I can do that. Presumably those methods should never be called to begin with if custom emojis are disabled

Reply:
Actually, I'm going to leave this as-is. Most of the other emoji methods in the `api` and `app` layers duplicate these permissions checks, and there's even some repeated checks like `app.CreateEmoji` which calls `app.UploadEmojiImage`, both of which check that setting. I don't want to risk accidentally breaking something by removing a check that's expected from somewhere else (like the bulk importer or something), and I'd rather keep the methods consistent with each other than to have one be "correct" and the others not